core/src/main/scala/reftree/diagram/Animation.scala

```diff
@@ -83,6 +83,32 @@ object Animation {
   /** Create an animation builder from a sequence of starting values */
   def startWithSequence[A: ToRefTree](start: Seq[A]) = Builder(start.toVector)
 
+  private implicit class TakeUntilStream[T](s: Stream[T]) {
+    def takeWhileInclusive(p: T => Boolean): Stream[T] =
+      if (!s.isEmpty && p(s.head)) Stream.cons(s.head, s.tail takeWhileInclusive p)
+      else Stream.cons(s.tail.head, Stream.empty)
+  }
+
+  /** Create an animation builder from a seed until a fixpoint (f(x) == x) is reached */
+  def iterateUntilFixPoint[A: ToRefTree](seed: A)(iteration: (A => A))(
+    max: Int = 10) =
+    Builder(
+      Stream
+        .iterate(seed)(iteration)
+        .take(max)
+        .takeWhileInclusive(x => x != iteration(x))
+        .toVector)
+
+  /** Create an animation builder from a seed until a a given predicate yields true */
+  def iterateUntil[A: ToRefTree](seed: A)(iteration: (A => A))(
```

Comment:
I agree about the following the while/until. For the use case -- I could easily see a lot of animations that should run until some specific state, i.e. keep running until my list has some size, until my graph cannot be reduced any more, until my list is sorted, until whatever.
I'm sure some people could have a "while" case as well. Should we add both?